base_conditional_image/models/abstract_conditional_image.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class ConditionalImageConsumerMixin(models.AbstractModel):
+    _name = "conditional.image.consumer.mixin"
+    _description = "Mixin for conditional images consumers"
+    _inherit = "image.mixin"
+
+    image_1920 = fields.Image(compute="_compute_images", store=False, readonly=True)
+    image_1024 = fields.Image(compute="_compute_images", store=False, readonly=True)
+    image_512 = fields.Image(compute="_compute_images", store=False, readonly=True)
+    image_256 = fields.Image(compute="_compute_images", store=False, readonly=True)
+    image_128 = fields.Image(compute="_compute_images", store=False, readonly=True)
+
+    def evaluate_image_selector(self, conditional_image):
+        self.ensure_one()
+        if (
+            conditional_image.company_id == self.company_id
+            or self.company_id
+            and not conditional_image.company_id
+        ):
+            return bool(safe_eval(
+                conditional_image.selector or "True",
+                {"object": self}
+            ))
+        return False
```

Comment:
AFAIS if you don't have a value for `selector` you don't need any evaluation.

```suggestion
        if (
            conditional_image.company_id == self.company_id
            or self.company_id
            and not conditional_image.company_id
        ):
            selector = conditional_image.selector.strip() if conditional_image.selector else ""
            if not selector:
                return True
            return bool(safe_eval(selector, {"object": self}))
        return False
```

Reply:
Done, sligthly differently ;)